docling_core/experimental/idoctags.py

```diff
@@ -1,199 +1,933 @@
+            IDocTagsAttributeKey.SELECTED: {
+                IDocTagsAttributeValue.TRUE,
+                IDocTagsAttributeValue.FALSE,
+            }
+        },
+        IDocTagsToken.INLINE: {
+            IDocTagsAttributeKey.CLASS: {
+                IDocTagsAttributeValue.FORMULA,
+                IDocTagsAttributeValue.CODE,
+                IDocTagsAttributeValue.PICTURE,
+            }
+        },
+        IDocTagsToken.FLOATING_GROUP: {
+            IDocTagsAttributeKey.CLASS: {
+                IDocTagsAttributeValue.DOCUMENT_INDEX,
+                IDocTagsAttributeValue.TABLE,
+                IDocTagsAttributeValue.PICTURE,
+                IDocTagsAttributeValue.FORM,
+                IDocTagsAttributeValue.CODE,
+            }
+        },
+        # Other attributes (e.g., level, type, id) are not enumerated here
+    }
+
+    ALLOWED_ATTRIBUTE_RANGE: ClassVar[
+        dict[IDocTagsToken, dict["IDocTagsAttributeKey", tuple[int, int]]]
+    ] = {
+        # Geometric: value in [0, res]; resolution optional.
+        # Keep conservative defaults aligned with existing usage.
+        IDocTagsToken.LOCATION: {
+            IDocTagsAttributeKey.VALUE: (0, DOCTAGS_RESOLUTION),
+            IDocTagsAttributeKey.RESOLUTION: (DOCTAGS_RESOLUTION, DOCTAGS_RESOLUTION),
+        },
+        # Temporal components
+        IDocTagsToken.HOUR: {IDocTagsAttributeKey.VALUE: (0, 99)},
+        IDocTagsToken.MINUTE: {IDocTagsAttributeKey.VALUE: (0, 59)},
+        IDocTagsToken.SECOND: {IDocTagsAttributeKey.VALUE: (0, 59)},
+        IDocTagsToken.CENTISECOND: {IDocTagsAttributeKey.VALUE: (0, 99)},
+        # Levels (N ≥ 1)
+        IDocTagsToken.HEADING: {IDocTagsAttributeKey.LEVEL: (1, 6)},
+        IDocTagsToken.FORM_HEADING: {IDocTagsAttributeKey.LEVEL: (1, 6)},
+        IDocTagsToken.SECTION: {IDocTagsAttributeKey.LEVEL: (1, 6)},
+        # Continuation markers (id length constraints)
+        IDocTagsToken.THREAD: {IDocTagsAttributeKey.ID: (1, 10)},
+        IDocTagsToken.H_THREAD: {IDocTagsAttributeKey.ID: (1, 10)},
+    }
+
+    # Self-closing tokens map
+    IS_SELFCLOSING: ClassVar[dict[IDocTagsToken, bool]] = {
+        IDocTagsToken.PAGE_BREAK: True,
+        IDocTagsToken.TIME_BREAK: True,
+        IDocTagsToken.LOCATION: True,
+        IDocTagsToken.HOUR: True,
+        IDocTagsToken.MINUTE: True,
+        IDocTagsToken.SECOND: True,
+        IDocTagsToken.CENTISECOND: True,
+        IDocTagsToken.BR: True,
+        # OTSL structural tokens are emitted as self-closing markers
+        IDocTagsToken.FCEL: True,
+        IDocTagsToken.ECEL: True,
+        IDocTagsToken.CHED: True,
+        IDocTagsToken.RHED: True,
+        IDocTagsToken.CORN: True,
+        IDocTagsToken.SROW: True,
+        IDocTagsToken.LCEL: True,
+        IDocTagsToken.UCEL: True,
+        IDocTagsToken.XCEL: True,
+        IDocTagsToken.NL: True,
+        # Continuation markers
+        IDocTagsToken.THREAD: True,
+        IDocTagsToken.H_THREAD: True,
+    }
```

Comment:
This seems to include only the self-closing ones (i.e. value is always `True`), so why not just make it a `set`?